src/routes/dashboard/containers/DashboardContainer.js

```diff
@@ -66,7 +69,16 @@ const mapStateToProps = (state: StateType, ownProps) => {
   }
 }
 
-// $FlowFixMe
 const themed = withTheme(Dashboard)
-// $FlowFixMe connect()
-export default withRouteCleaner(connect(mapStateToProps, mapDispatchToProps)(withError(translate('dashboard')(themed))))
+export default compose([
+  withRouteCleaner,
+  connect((state: StateType): { invalidLanguage: boolean } => {
+    const languages = state.cityContent.languages
+    const language = state.cityContent.language
+    return { invalidLanguage: !!languages && !languages.find(languageModel => languageModel.code === language) }
```

Comment:
what's the purpose here?

Reply:
He checks in the state if the currentLanguage is available and sets the `invalidLanguage` prop accordingly. In the `branch` (next line), he checks whether he should display a LanguageNotAvailableContainer or not.